website/project/model.py

```diff
@@ -140,17 +140,27 @@ def validate_comment_reports(value, *args, **kwargs):
 
 class Comment(GuidStoredObject):
 
+    OVERVIEW = "node"
+    FILES = "files"
+    WIKI = "wiki"
+
     _id = fields.StringField(primary=True)
 
     user = fields.ForeignField('user', required=True, backref='commented')
     node = fields.ForeignField('node', required=True, backref='comment_owner')
     target = fields.AbstractForeignField(required=True, backref='commented')
+    # The file/wiki/overview's page that the comment is for
+    root_target = fields.AbstractForeignField(backref='comment_target')
 
     date_created = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow)
     date_modified = fields.DateTimeField(auto_now=datetime.datetime.utcnow)
     modified = fields.BooleanField()
 
     is_deleted = fields.BooleanField(default=False)
+    # Whether the original file/wiki is deleted
+    is_hidden = fields.BooleanField(default=False)
```

Comment:
No, and I haven't thought about this!
I don't know how we can retrieve a deleted node, but you're welcome to add a line to hide the comments when the node is deleted... Just note that the file comments are never hidden in the backend and should never be